web/libs/editor/src/tags/object/AudioUltra/view.tsx

```diff
@@ -180,49 +165,80 @@ const AudioUltraView: FC<AudioUltraProps> = ({ item }) => {
-          onPlay={() => controls.setPlaying(true)}
-          onPause={() => controls.setPlaying(false)}
-          allowFullscreen={false}
-          onVolumeChange={(vol) => controls.setVolume(vol)}
-          onStepBackward={() => {
-            waveform.current?.seekBackward(NORMALIZED_STEP);
-            waveform.current?.syncCursor();
-          }}
-          onStepForward={() => {
-            waveform.current?.seekForward(NORMALIZED_STEP);
-            waveform.current?.syncCursor();
-          }}
-          onPositionChange={(pos) => {
-            waveform.current?.seek(pos);
-            waveform.current?.syncCursor();
-          }}
-          onSpeedChange={(speed) => controls.setRate(speed)}
-          onZoom={(zoom) => controls.setZoom(zoom)}
-          amp={controls.amp}
-          onAmpChange={(amp) => controls.setAmp(amp)}
-          mediaType="audio"
-          toggleVisibility={(layerName: string, isVisible: boolean) => {
-            if (waveform.current) {
-              const layer = waveform.current?.getLayer(layerName);
-
-              if (layer) {
-                layer.setVisibility(isVisible);
-              }
+      <Controls
+        position={controls.currentTime}
+        playing={controls.playing}
+        volume={controls.volume}
+        speed={controls.rate}
+        zoom={controls.zoom}
+        duration={controls.duration}
+        onPlay={() => controls.setPlaying(true)}
+        onPause={() => controls.setPlaying(false)}
+        allowFullscreen={false}
+        onVolumeChange={(vol) => controls.setVolume(vol)}
+        onStepBackward={() => {
+          waveform.current?.seekBackward(NORMALIZED_STEP);
+          waveform.current?.syncCursor();
+        }}
+        onStepForward={() => {
+          waveform.current?.seekForward(NORMALIZED_STEP);
+          waveform.current?.syncCursor();
+        }}
+        onPositionChange={(pos) => {
+          waveform.current?.seek(pos);
+          waveform.current?.syncCursor();
+        }}
+        onSpeedChange={(speed) => controls.setRate(speed)}
+        onZoom={(zoom) => controls.setZoom(zoom)}
+        amp={controls.amp}
+        onAmpChange={(amp) => controls.setAmp(amp)}
+        mediaType="audio"
+        toggleVisibility={(layerName: string, isVisible: boolean) => {
+          if (waveform.current) {
+            const layer = waveform.current?.getLayer(layerName);
+
+            if (layer) {
+              layer.setVisibility(isVisible);
             }
-          }}
-          layerVisibility={controls.layerVisibility}
-        />
-      </TimelineContextProvider>
+          }
+        }}
+        layerVisibility={controls.layerVisibility}
+      />
     </Block>
   );
 };
 
-export const AudioUltra = observer(AudioUltraView);
+const AudioUltraWithSettings: FC<AudioUltraProps> = ({ item }) => {
+  const [settings, setSettings] = usePersistentJSONState<TimelineSettings>("ls:audio-tag:settings", {
+    playpauseHotkey: "audio:playpause",
```

Comment:
Since it is persistent, I am starting to have doubts about whether it is possible to store keyboard shortcuts here.
It looks like it would be hard to change hotkeys in the future.

Reply:
Yeah, I don't even like the idea of this hotkey to be a setting like this and it's not even possible to change it.

Reply:
Added a comment about this hotkey, thank you!